src/hyppo/freeze.asm

```diff
@@ -64,6 +64,12 @@ unfreeze_next_region:
 
         jsr unfreeze_load_region
 
+        ;; Re-enable M65 IO in case we wrote over the key register during the region unfreeze
```

Comment:
@gardners If you have any time, please have a look on this, it's very interesting, that it seems removing $D02F (KEY register) writing in hypervisor mode should be totally unnecessary (and remanent of the past) but still, it seems we cannot remove at least _all_ of them without some ill effect ... Thanks. I find this important as it may mean some deeper problem somewhere (in VHDL?) that KEY register writing in hypervisor mode still have _some_ effect. Or my knowledge here is not enough, the other possibility ;)

Reply:
The reason is that the _CPU_ ignores the state of the key register while in Hypervisor Mode, but the flag is still changed. This may actually indicate that the KEY register setting is actually not possible to freeze in its current design at all, which is the real issue. Thus the unfreeze enabling VIC-IV mode is masking the problem somewhat, but is also not a perfect solution.

Reply:
@gardners Ah, ok. But then, as far as I know at least, returning from hypervisor mode, I/O mode is set from $D652 anyway.  Thus setting $D02F in hypervisor mode has no effect (even if it's flagged) _and_ anyway after leaving hypervisor mode, I/O mode is set according to $D652 (which is btw used to "backup" I/O mode on entering hypervisor mode, at least IMHO). But as far as I see hyppo does not write this register at all, only in task.asm, here:

    task_set_c64_memorymap:
        [...]
        lda #$00
        sta hypervisor_iomode    ;; C64 IO map

So maybe I know wrongly (and I did this Xemu in the wrong way) but afaik, when hyppo does freeze/unfreeze in hypervisor mode it should store $D652 (I/O mode used before freezing) and should write $D652 at unfreeze, so when returning from hypervisor mode have the desired I/O made active.

Reply:
@lgblgblgb How about for now, I create another ticket so that one of us can dig deeper into this later. Just feel like we share this potential solution with end users quickly, and leave deeper investigation for that ticket.

Reply:
Ok, I've made this ticket for the matter:
- https://github.com/MEGA65/mega65-core/issues/569